oneflow/api/python/framework/autocast.cpp

```diff
@@ -48,16 +50,23 @@ class AutoCastMode {
     increase_nested_count();
     autocast::set_enabled(enabled);
     autocast::set_autocast_cache_enabled(cache_enabled);
-    if (device_type == "cpu") {
-      autocast::set_autocast_device_type(kCPU);
-      autocast::set_autocast_dtype(dtype);
-      autocast::set_autocast_cpu_dtype(dtype);
-    } else if (device_type == "cuda") {
-      autocast::set_autocast_device_type(kCUDA);
-      autocast::set_autocast_dtype(dtype);
-      autocast::set_autocast_gpu_dtype(dtype);
-    } else {
-      THROW(RuntimeError) << "User specified autocast device_type must be 'cuda' or 'cpu'";
+    auto device_type = ep::DeviceManagerRegistry::GetDeviceTypeByDeviceTypeName(device_name);
```

Comment:
Check if `device_type` is valid, and then we don't need `switch (device_type)` anymore.